internal/graphapi/controlextended.resolvers.go

```
@@ -37,96 +34,9 @@ func (r *mutationResolver) CreateControlsByClone(ctx context.Context, input *mod
-				mappedControlIDs = append(mappedControlIDs, mc.ID)
-			}
-
-			// if the control has subcontrols, we need to create them as well
-			subcontrols := make([]*generated.CreateSubcontrolInput, len(control.Edges.Subcontrols))
-			for j, subcontrol := range control.Edges.Subcontrols {
-				subcontrols[j] = &generated.CreateSubcontrolInput{
-					Tags:                   subcontrol.Tags,
-					RefCode:                subcontrol.RefCode,
-					Description:            &subcontrol.Description,
-					Source:                 &subcontrol.Source,
-					ControlID:              res.ID,
-					ControlType:            &subcontrol.ControlType,
-					Category:               &subcontrol.Category,
-					CategoryID:             &subcontrol.CategoryID,
-					Subcategory:            &subcontrol.Subcategory,
-					MappedCategories:       subcontrol.MappedCategories,
-					AssessmentObjectives:   subcontrol.AssessmentObjectives,
-					ControlQuestions:       subcontrol.ControlQuestions,
-					ImplementationGuidance: subcontrol.ImplementationGuidance,
-					ExampleEvidence:        subcontrol.ExampleEvidence,
-					References:             subcontrol.References,
-					MappedControlIDs:       mappedControlIDs,
-					Status:                 &enums.ControlStatusPreparing,
-				}
-			}
-
-			if _, err := r.bulkCreateSubcontrol(ctx, subcontrols); err != nil {
-				return nil, parseRequestError(err, action{action: ActionCreate, object: "control"})
-			}
-		}
-	}
-
-	// get the created controls
-	query, err := withTransactionalMutation(ctx).Control.Query().Where(control.IDIn(createdControlIDs...)).CollectFields(ctx)
+	createdControls, err := r.cloneControls(ctx, existingControls, input.ProgramID, false)
 	if err != nil {
-		return nil, parseRequestError(err, action{action: ActionCreate, object: "control"})
-	}
-
-	createdControls, err := query.All(ctx)
-	if err != nil {
-		return nil, parseRequestError(err, action{action: ActionCreate, object: "control"})
+		return nil, err
```

Resolver should return the parsed error instead of a generic error
```suggestion
		return nil, parseRequestError(err, action{action: ActionCreate, object: "control"})
```